### PR TITLE
Add ImageLoader.clearMemory(key) to clear a single memory cache item

### PR DESCRIPTION
**Pull request by Goddchen**
As discussed/requested by a user on Slack. Being able to clear a single item by its key from memory cache.

### CODE REVIEW

**Comment by colinrtwhite** [comment]
Thanks for adding this. I want to spend a bit more time thinking about the design for the moment. This is a breaking change to the base `ImageLoader` interface which is pretty major.

**Comment by Goddchen** [comment]
@Javieraeros merge conflicts resolved 👍

**Comment by mario** [comment]
What do we do with this PR @colinrtwhite ?

**Comment by Goddchen** [comment]
Merge? 😂

File: coil-base/src/main/java/coil/ImageLoader.kt [hunk 1]
```diff
@@ -59,6 +59,11 @@ interface ImageLoader {
      */
     fun clearMemory()
 
+    /**
+     * Remove a single item from memory cache.
+     */
+    fun clearMemory(key: String)
```
Comment (colinrtwhite) [inline coil-base/src/main/java/coil/ImageLoader.kt]: Can we rename this to `invalidate`? The behaviour of this function is different enough from `clearMemory()` that it should probably have its own name. `clearMemory()` empties the bitmap pool as well as the memory cache whereas `clearMemory(key)` only removes a specific memory cache entry.

**Comment by colinrtwhite** [comment]
Sorry for the delay on this! @Goddchen left one comment. I'm planning to do a quick `0.9.3` release soon, but after that release + the PR feedback, let's merge this in :).

**Comment by Goddchen** [comment]
@colinrtwhite merged coil/master, resolved conflicts and renamed the method as requested. I think it can be merged now 👍

**Comment by colinrtwhite** [comment]
@Goddchen Awesome! Thanks for the contribution.

**Pull request merged**